bql/semantic/expression.go

```diff
@@ -84,85 +84,41 @@ func (o OP) String() string {
 // evaluationNode represents the internal representation of one expression.
 type evaluationNode struct {
 	op OP
+
 	lB string
 	rB string
 }
 
-// comparisonForNodeLiteral represents the internal representation of a expression of comparison between a binding and a node literal.
-type comparisonForNodeLiteral struct {
-	op OP
-
-	lB  string
-	rNL string
-}
-
-func (e *comparisonForNodeLiteral) Evaluate(r table.Row) (bool, error) {
-	// Binary evaluation
-	eval := func() (*table.Cell, error) {
-		var (
-			eL *table.Cell
-			ok bool
-		)
-		eL, ok = r[e.lB]
-		if !ok {
-			return nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.lB, r)
-		}
-		return eL, nil
-	}
-
-	cs := func(c *table.Cell) string {
-		if c.L != nil {
-			return strings.TrimSpace(c.L.ToComparableString())
-		}
-		return strings.TrimSpace(c.String())
+// format cell to a trimmed comparable string
+func formatCell(c *table.Cell) string {
+	if c.L != nil {
+		return strings.TrimSpace(c.L.ToComparableString())
 	}
+	return strings.TrimSpace(c.String())
+}
 
-	eL, err := eval()
-	if err != nil {
-		return false, err
-	}
-	csEL, csER := cs(eL), strings.TrimSpace(e.rNL)
-	switch e.op {
-	case EQ:
-		return reflect.DeepEqual(csEL, csER), nil
-	case LT:
-		return cs(eL) < csER, nil
-	case GT:
-		return csEL > csER, nil
-	default:
-		return false, fmt.Errorf("boolean evaluation require a boolean operation; found %q instead", e.op)
+// binary evaluation - get value for binding in given row
```

Comment:
Extracting this so that it is no longer an anonymous function causes the name to be too broad.
I would suggest to rename it to explain the specific value it is extracting, e.g., cell cellFromTable.

Reply:
I agree, I have just renamed it, thank you Kloss.